deploy/deploy.ts

```diff
@@ -57,3 +42,64 @@ async function deployAll(
     });
     return voucherHubAddress;
 }
+
+/**
+ * Get deployment config according to chain.
+ */
+async function getDeploymentConfig() {
+    let [pocoAddress, upgraderAddress, managerAddress, minterAddress]: string[] = [];
+    const chainId = (await ethers.provider.getNetwork()).chainId;
+    console.log('ChainId:', chainId);
+    // Read default config of the target chain.
+    const chainConfig = deploymentConfig[chainId.toString()];
+    if (chainConfig) {
+        pocoAddress = chainConfig.pocoAddress;
+        upgraderAddress = chainConfig.roles.upgraderAddress;
+        managerAddress = chainConfig.roles.managerAddress;
+        minterAddress = chainConfig.roles.minterAddress;
+    }
+    // Override config if required.
+    if (process.env.IEXEC_POCO_ADDRESS) {
+        pocoAddress = process.env.IEXEC_POCO_ADDRESS;
+    }
+    if (process.env.IEXEC_VOUCHER_UPGRADER_ADDRESS) {
+        upgraderAddress = process.env.IEXEC_VOUCHER_UPGRADER_ADDRESS;
+    }
+    if (process.env.IEXEC_VOUCHER_MANAGER_ADDRESS) {
+        managerAddress = process.env.IEXEC_VOUCHER_MANAGER_ADDRESS;
+    }
+    if (process.env.IEXEC_VOUCHER_MINTER_ADDRESS) {
+        minterAddress = process.env.IEXEC_VOUCHER_MINTER_ADDRESS;
+    }
```

Comment:
```suggestion
    const chainId = (await ethers.provider.getNetwork()).chainId;
    console.log('ChainId:', chainId);
    // Read config file of the target chain or read env
    const configFile = deploymentConfig[chainId.toString()];
    const configFileRoles = configFile?.roles;
    let pocoAddress = configFile?.pocoAddress || process.env.IEXEC_POCO_ADDRESS || '';
    let upgraderAddress =
        configFileRoles?.upgraderAddress || process.env.IEXEC_VOUCHER_UPGRADER_ADDRESS || '';
    let managerAddress =
        configFileRoles?.upgraderAddress || process.env.IEXEC_VOUCHER_MANAGER_ADDRESS || '';
    let minterAddress =
        configFileRoles?.minterAddress || process.env.IEXEC_VOUCHER_MINTER_ADDRESS || '';
```

Reply:
This way, if (for e.g.) `pocoAddress` is defined in config file it will not be overridden by env variable.

Reply:
Ok alright!